docs/en/api/FS-API.md

```diff
@@ -187,23 +212,66 @@ Users can override the default policy class in the
 [configuration file]({{ '/en/operation/Configuration.html' | relativize_url }}) at property
 `alluxio.user.block.write.location.policy.class`. The built-in policies include:
 
-* **LocalFirstPolicy (alluxio.client.block.policy.LocalFirstPolicy)**
+* [LocalFirstPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/LocalFirstPolicy.html)
+
+  **This is the default policy.**
+
+  > A policy that returns the local worker first, and if the local worker doesn't
+  > exist or have enough availability, will select the nearest worker from the active
+  > workers list with sufficient availability.
+  >
+  > The calculation of which worker gets selected is done for each block write.
+
+* [LocalFirstAvoidEvictionPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/LocalFirstAvoidEvictionPolicy.html)
+
+  This is the same as `LocalFirstPolicy` with the following addition:
 
-    Returns the local worker first, and if it does not have enough capacity of a block,
-    randomly picks a worker from the active workers list. This is the default policy.
+  > A policy that returns the local worker first, and if the local worker doesn't
+  > exist or have enough availability, will select the nearest worker from the active
+  > workers list with sufficient availability.
+  >
+  > The calculation of which worker gets selected is done for each block write.
+  >
+  > The PropertyKey `USER_FILE_WRITE_AVOID_EVICTION_POLICY_RESERVED_BYTES`
+  > (alluxio.user.block.avoid.eviction.policy.reserved.size.bytes)
+  > is used as buffer space on each worker when calculating available space
+  > to store each block.
 
-* **MostAvailableFirstPolicy (alluxio.client.block.policy.MostAvailableFirstPolicy)**
+* [MostAvailableFirstPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/MostAvailableFirstPolicy.html)
 
-    Returns the worker with the most available bytes.
+  > A policy that returns the worker with the most available bytes.
+  >
+  > The policy returns null if no worker is qualified.
```

Comment:
I think if this policy returns null, the client will select someone else. So from a user perspective, they don't need to know that the method can return null (nor should they worry about it). Let's just mention the behavior, which is (correct me if i'm wrong): local worker not available -> select random or select nearest?

Reply:
I asked in the Dev channel on Slack and it seems no one is familiar enough with this part of the code to say what exactly happens when the returned worker is `null`. When I tested it appears that a worker is selected at random.